stetho/src/main/java/com/facebook/stetho/common/android/HandlerUtil.java

```diff
@@ -11,38 +11,104 @@
 
 import android.os.Handler;
 import android.os.Looper;
+import com.facebook.stetho.common.Util;
+
+import java.util.concurrent.Callable;
 
 public final class HandlerUtil {
   private HandlerUtil() {
   }
 
-  public static boolean postAndWait(Handler handler, final Runnable r) {
-    if (Looper.myLooper() == handler.getLooper()) {
-      r.run();
-      return true;
+  /**
+   * Checks whether the current thread is the same thread that the {@link Handler} is associated
+   * with.
+   * @return true if the current thread is the same thread that the {@link Handler} is associated
+   * with; otherwise false.
+   */
+  public static boolean checkAccess(Handler handler) {
```

Comment:
WPF uses this extensively and I found it to work quite well: https://msdn.microsoft.com/en-us/library/system.windows.threading.dispatcherobject.checkaccess%28v=vs.110%29.aspx

Perhaps it would work better with a name like checkThreadAccess(). It is a lot less code when you have to repeat the call in many places; it's intended to be used non-sparingly in order to enforce that code is running on the right thread. It's not really different in principle than something like Util.throwIfNull().